### PR TITLE
fix azure devops pipeline

### PR DESCRIPTION
**Pull request by davidpolaniaac**
Before submitting a pull request, please read
https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing

## Description
Fix SonarQube properties for correct static analysis of source code. 
Updates the versions of the tasks in the pipeline. 
Adds pipeline schema reference as code and enhancements to the pipeline.

## Category
- [ ] Feature
- [x] Fix

## Checklist
- [x] The pull request is complete according to the [guide of contributing](https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing)
- [x] Automated tests are written
- [x] The documentation is up-to-date
- [x] The pull request has a descriptive title that describes what has changed, and provides enough context for the changelog
- [ ] If the pull request has changed structural files, you have implemented an UpgradeAction according to the [guide](https://github.com/bancolombia/scaffold-clean-architecture/wiki/Contributing#upgradeaction)

### CODE REVIEW

**Comment by davidpolaniaac** [comment]
@santitigaga

File: src/main/resources/pipeline/azure/azure.yaml.mustache [hunk 1]
```diff
@@ -1,79 +1,68 @@
...
-      - task: SonarQubePrepare@4
-        displayName: 'Prepare analysis on SonarQube'
-        inputs:
-          SonarQube: SonarQube
-          scannerMode: Other
-          extraProperties: |
-            sonar.projectKey=$(Build.Repository.Name)
-            sonar.projectName=$(Build.Repository.Name)
-            sonar.project.version=$(Build.BuildNumber)
-            sonar.branch.name=$(Build.SourceBranchName)
```
Comment (santitigaga) [inline src/main/resources/pipeline/azure/azure.yaml.mustache:48]: What happened with `sonar.branch.name` this is unnecessary?

**Review by santitigaga — COMMENTED** [review]
Great!!!

File: src/main/resources/pipeline/azure/azure.yaml.mustache (hunk 1 above)
Comment (davidpolaniaac) [reply src/main/resources/pipeline/azure/azure.yaml.mustache:48]: When a build is run on a branch of project, the extension automatically configures the analysis to be pushed to the relevant project branch.

**Pull request merged**